### ISSUE TITLE
Errata for Bioinformatics Data Skills

### ISSUE THREAD

**Issue by haruosuz** [vsbuffalo/bds-files#2]
# Errata for Bioinformatics Data Skills

[Confirmed Errata | O'Reilly Media](http://www.oreilly.com/catalog/errata.csp?isbn=9781449367374)

----------

## Computer environment
I’ve run examples in this book on Mac OS X 10.9.5 in which I installed Python 3.4.3 and R version 3.2.2.

```
Python 3.4.3 |Anaconda 2.3.0 (x86_64)| (default, Mar  6 2015, 12:07:41) 
[GCC 4.2.1 (Apple Inc. build 5577)] on darwin


> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-apple-darwin13.4.0 (64-bit)
Running under: OS X 10.9.5 (Mavericks)

other attached packages:
[1] ggplot2_1.0.1
```

----------

## Chapter 3. Remedial Unix Shell

-----
(Pages 38 and 45)

Doug McIlory

*should be*

Doug McIlroy

https://en.wikipedia.org/wiki/Douglas_McIlroy

-----
(Page 49) 

can can combine pipes and redirects easily: 

*should be*

can combine pipes and redirects easily:

-----
(Page 53) 

The shell operator && executes subsequent commands only if previous commands have completed with a nonzero exit status:

*should be*

The shell operator && executes subsequent commands only if previous commands have completed with a zero exit status:

----------

## Chapter 4. Working with Remote Machines

----------

## Chapter 5. Git for Scientists

-----
(Page 89)

our remote repository to a local directory named zmay-snps-barbara/.

*should be*

our remote repository to a local directory named zmays-snps-barbara/.

-----
(Page 89)

Now, in our original zmay-snps/ local

*should be*

Now, in our original zmays-snps/ local

----------

## Chapter 6. Bioinformatics Data

-----
(Page 113)

The option -a enables wrsync’s archive mode, 

*should be*

The option -a enables rsync’s archive mode, 

-----
 (Page 122)

	$ shasum cf5bb5f8bda2803410bb04b708bff59cb575e379 Mus_musculus.GRCm38.74.gtf.gz

*should be*

	$ shasum Mus_musculus.GRCm38.74.gtf.gz
	cf5bb5f8bda2803410bb04b708bff59cb575e379  Mus_musculus.GRCm38.74.gtf.gz

----------

----------

## Chapter 12. Bioinformatics Shell Scripting, Writing Pipelines, and Parallelizing Tasks

-----
(Page 403-404)

	$ test -r some_file.txt ; echo $? $ is this file readable?
	$ test -w some_file.txt ; echo $? $ is this file writable?

*should be*

	$ test -r some_file.txt ; echo $? # is this file readable?
	$ test -w some_file.txt ; echo $? # is this file writable?

-----
(Page 404-405)

`./script.sh` printed "./script.sh: line 6: $1: unbound variable".

-----
(Page 414)

	find seqs -type f "!" -name "zmaysC*fastq"

*printed "-bash: !: event not found" and should thus be*

	find seqs -type f ! -name "zmaysC*fastq"

-----
(Page 414)

(we are still in the zmays/data directory):

should be

(we are still in the zmays-snps/data directory):

-----
(Page 415)

	find seqs -type f "!" -name "zmaysC*fastq" -and "!" -name "*-temp*"

*should be*

	find seqs -type f ! -name "zmaysC*fastq" -and ! -name "*-temp*"

----------

## Chapter 13. Out-of-Memory Approaches: Tabix and SQLite

-----

Using Tabix Version: 1.2.1 on Mac OS X 10.9.5, `man tabix` printed "No manual entry for tabix"

----------

**Comment by mcivelek** [comment]
For gwascat.db, I get the following error in both Unix and Windows: Error: file is encrypted or is not a database Do you know what might be the problem?

**Comment by vsbuffalo** [comment]
Could you send me the command you're using?

**Comment by mcivelek** [comment]
sqlite3 gwascat.db
sqlite> .schema
Error: file is encrypted or is not a database

**Comment by vsbuffalo** [comment]
It's very likely an issue with the way it downloaded. It seems to work for me with:

```
$ wget http://raw.githubusercontent.com/vsbuffalo/bds-files/master/chapter-13-out-of-memory/gwascat.db
$ sqlite3 ./gwascat.db
SQLite version 3.8.5 2014-08-15 22:37:57
Enter ".help" for usage hints.
sqlite> .schema
CREATE TABLE gwascat(
id integer PRIMARY KEY NOT NULL,
dbdate text,
pubmedid integer,
author text,
date text,
[...]
```

You might also try uninstalling and re-installing sqlite3. It's surprising how often this works.

In the future, could you create a new Github issue? It helps keep the issues separate.

**Comment by mcivelek** [comment]
yes, it was the way it downloaded. When I used the URL in your reply it worked fine. Thank you.

**Comment by NamLQ** [comment]
page 161.

In the original code, we subtract 1 from the start position to convert to BED format because of zero-indexing. 

`$ awk '!/^#/ { print $1 "\t" $4-1 "\t" $5 }' Mus_musculus.GRCm38.75_chr1.gtf | head -n 3`

Should we also subtract the end position like this?

`$ awk '!/^#/ { print $1 "\t" $4-1 "\t" $5-1 }' Mus_musculus.GRCm38.75_chr1.gtf | head -n 3`
